napari/layers/base/base.py

```diff
@@ -197,28 +260,14 @@ def position(self, position):
         self._position = position
         self._update_coordinates()
 
-    def _update_coordinates(self):
-        """Insert the cursor position (x, y) into the correct position in the
-        tuple of indices and update the cursor coordinates.
-        """
-        if self._node.canvas is not None:
-            transform = self._node.canvas.scene.node_transform(self._node)
-            position = transform.map(list(self.position))[
-                : len(self.dims.displayed)
-            ]
-            position = position[::-1]
-        else:
-            position = [0] * len(self.dims.displayed)
-
-        coords = list(self.dims.indices)
-        for d, p in zip(self.dims.displayed, position):
-            coords[d] = p
-        self.coordinates = tuple(coords)
-
     def _update_dims(self):
         """Updates dims model, which is useful after data has been changed."""
         range = self._get_range()
-        self.dims.ndim = len(range)
+        if len(range) != self.dims.ndim:
```

Comment:
Can you add a comment explaining this condition? Under what circumstances will this be true?

Additionally, `range` shadows the built-in — it's not a *huge* deal, but I would prefer if we avoided that. `curr_range`? `self_range`?

Reply:
fixed with `curr_range`